ui/src/widgets/ui-form/UIForm.vue

```diff
@@ -38,17 +38,28 @@ export default {
         id: { type: String, required: true },
         props: { type: Object, default: () => ({}) }
     },
-    setup (props) {
-        useDataTracker(props.id)
-    },
     data () {
         return {
             input: {},
-            isValid: null
+            isValid: null,
+            dynamic: {
+                label: null,
+                options: null
+            }
         }
     },
     computed: {
-        ...mapState('data', ['messages'])
+        ...mapState('data', ['messages']),
+        label: function () {
+            return this.dynamic.label || this.props.label
```

Comment:
This will not permit user to have an empty label. 

As per other nodes (ui dropdown, ui slider), we should allow empty string) e.g.

`return this.dynamic.label !== null ? this.dynamic.label : this.props.label`

That is assuming the default value is `null`

This really does back up the need for common/shared helper methods as was mentioned in another recent PR in similar situation.